# Dynamic Calculator: 1+2×3(4-5÷4)-(3÷5)

. Both the first and second methods are suitable for Jupyter Notebook.

. The first method is also suitable for Python . In jupiter we call it by using print()

. The second method is only suitable for Jupyter Notebook

# . First method

In [ ]:
step1 = 5/4
step2 = 4-step1
step3 = 3*step2
step4 = 2 *step3
step5 = 1+step4
step6 = step5-3/5


In [17]:
print(step1)
print(step2)
print(step3)
print(step4)
print(step5)
print(step6)
print("result=",step6)

1.25
2.75
8.25
16.5
17.5
16.9
result= 16.9


# . Description of 1st method:

   Why we use this?

    . debugging, or writing step-by-step calculations.

# Step 1:
We solve the division inside the parentheses:
5÷4=1.25
So now the expression inside the brackets becomes:
4−1.25

# Step2:
4 - step1
4−1.25=2.75
This completes the part inside the parentheses.

# Step3:
 3 * step2
3×2.75=8.25

#Step4:
2 * step3
This completes:2×3(4−5/4)

# Step5:
1 + step4
1+16.5=17.5
Now we've completed:
1+2×3(4−5/4)

# Step6:
step5 - 3 / 5
= 16.9



# . Second method:

In [20]:
from ipywidgets import interact, FloatSlider

def cal(a=1, b=2, c=3, d=4, e=5, f=4, g=3, h=5):
    # Expression: a + b*c*(d - e/f) - (g/h)
    result = a + b * c * (d - e / f) - (g / h)
    print(f'Result: {result:.4f}')

interact(cal,
         a=FloatSlider(value=1, min=0, max=10, step=0.1),
         b=FloatSlider(value=2, min=0, max=10, step=0.1),
         c=FloatSlider(value=3, min=0, max=10, step=0.1),
         d=FloatSlider(value=4, min=0, max=10, step=0.1),
         e=FloatSlider(value=5, min=0, max=10, step=0.1),
         f=FloatSlider(value=4, min=0.1, max=10, step=0.1),
         g=FloatSlider(value=3, min=0, max=10, step=0.1),
         h=FloatSlider(value=5, min=0.1, max=10, step=0.1))


interactive(children=(FloatSlider(value=1.0, description='a', max=10.0), FloatSlider(value=2.0, description='b…

<function __main__.cal(a=1, b=2, c=3, d=4, e=5, f=4, g=3, h=5)>

# Description of 2nd method :

	
Why we use this?

       it is suitable for interactive notebook environments.
	
	
	• interact: a function that turns a regular function into an interactive widget.automatically create sliders,layout etc. 
	• FloatSlider: a slider that allows  decimal (float) values.if we import floatslider then we will specify min,max and step. we mayuse another slider and so on.
• This is designed for Jupyter Notebooks.


• Define a function with any name. write 8 parameters and give then value according to the question.
• Prints the result of the expression.

• :.3f formats means after deciaml it puts three zeros.


Creates interactive sliders for each parameter:

	• value: given in question

	• min: minimum allowed value(lowest value is 0)

	• max: maximum allowed value(100,500……)
	
	• step: how much the slider changes when moved(0.1 means fine slider)
	
	• a = 1 → Yeh expression first number
(From 1 + ...)

	• b = 2 → 1st  multiplier
(From 2 × 3)

	• c = 3 → 2nd multiplier
(From 2 × 3)

	• d = 4 → Bracket first  number
(From 4 - ...)

	• e = 5 → Bracket division numerator
(From 5 ÷ 4)

	• f = 4 → Bracket division denominator
(Min = 0.1 to avoid divide by zero )

	• g = 3 → Final subtraction  numerator
(From - (3 / 5))

	• h = 5 → Final division  denominator
(Min = 0.1 to avoid divide by zero)


